index.js

```diff
@@ -242,6 +242,35 @@ function build (schema, options) {
   ajvInstance = new Ajv({ ...options.ajv, strictSchema: false, uriResolver: fastUri })
   ajvFormats(ajvInstance)
 
+  const validateDateTimeFormat = ajvFormats.get('date-time').validate
+  const validateDateFormat = ajvFormats.get('date').validate
+  const validateTimeFormat = ajvFormats.get('time').validate
+
+  ajvInstance.addKeyword({
+    keyword: 'fjs_date_type',
+    validate: (schema, date) => {
+      if (date && typeof date.toISOString === 'function') {
+        return true
+      }
+      if (schema === 'date-time') {
+        return validateDateTimeFormat(date)
+      }
+      if (schema === 'date') {
+        if (date && typeof date.format === 'function') {
```

Comment:
It can be not only Date type. Fjs supports different formats of date objects. moment lib for example

Reply:
The only example of using a non-standard `Date` object in the Readme is a `moment` example in which the `moment` instance has been stringified into an RFC 3339 string. It would be prohibitively difficult to support any other type of "date" object that is not a standard `Date` object. For example, look at https://js-joda.github.io/js-joda/manual/formatting.html. Not only are the formatting tokens completely different from the `moment` tokens, the `.format` function expects an object instead of a string. Even the `luxon` tokens are different from `moment` https://moment.github.io/luxon/#/formatting?id=table-of-tokens.

FJS should be looking for a standard `Date` object or a string.

Reply:
In that case oneOf anyOf will not work if it's a moment object.

Reply:
We don't only validate, we serialize date according to these rules.
https://github.com/fastify/fast-json-stringify/blob/074a120dcdaa97759f2df9890a3f9708a7e1cba0/index.js#L145

I also don't think it's a good idea, but it's already supported.

Reply:
I see that I missed where `.format` was used in https://github.com/fastify/fast-json-stringify/pull/195. That was a mistake. We should never have relied on that method, for all of the reasons I stated above and for https://github.com/fastify/fast-json-stringify/issues/428.

I'd say this is a good opportunity to fix that mistake.

Reply:
> Let's clarify it. Can we drop moment lib support?

I think the current tests should continue to pass in order to avoid a semver major release. But, yes, I'd say dropping direct `moment` support is a good idea. The section in the Readme detailing the support for `Dates` should be adhered to as discussed above.